bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/response/PatientResponse.java

```diff
@@ -208,4 +211,43 @@ public void serialize(Date date, JsonGenerator gen, SerializerProvider provider)
             gen.writeString(formattedDate);
         }
     }
+    
+    /**
+     * Localizes the address value of an address field value string.
+     * Eg. {"addressField" : "address.highStreet"} -> {"addressField" : "High Street"}
+     * 
+     * @param addressFieldValue The address field value JSON.
+     * @return The localized address field value JSON.
+     */
+    public static String localizeAddressFieldValue(String addressFieldValue) {
+    	try {
+    		HashMap<String, String> map = new ObjectMapper().readValue(addressFieldValue, HashMap.class);
+    		for (String addressField : map.keySet()) {
+    			String addressValue = Context.getMessageSourceService().getMessage(map.get(addressField));
+    			map.put(addressField, addressValue);
+    		}
+    		addressFieldValue = new ObjectMapper().writeValueAsString(map);
+    	} catch (Exception e) {} 
+    	return addressFieldValue;
+    }
+
+    /**
+     * Localizes the patient identifier types of the extra identifiers.
+     * Eg. {"pit.id1" : "98765321", "pit.id2" : "MRS-1234"} -> {"Primary ID" : "98765321", "Secondary ID" : "MRS-1234"} 
+     * 
+     * @param extraIdentifiers The extra identifiers JSON string.
+     * @return The localized extra identifiers JSON string.
+     */
+    public static String localizePatientIdentifierTypes(String extraIdentifiers) {
+    	try {
+    		HashMap<String, String> map = new ObjectMapper().readValue(extraIdentifiers, HashMap.class);
+    		for (String pit : map.keySet()) {
+    			String l10nPit = Context.getMessageSourceService().getMessage(pit);
+    			if (!pit.equals(l10nPit)) {
+    				extraIdentifiers = extraIdentifiers.replaceAll(pit, l10nPit);
+    			}
+    		}
+    	} catch (Exception e) {} 
```

Comment:
No exception handling?

Reply:
Yes I should WARN-log this yes sure. Thank you.

But I don't want to throw any exception. The point is that if anything goes wrong with any of those localization methods, they should revert to the behavior "as if nothing was attempted". Hence the apparent silent failing.
The bottom line is that localization in fact does _not_ happen by default.